thunx-pdp-opa/src/main/java/com/contentgrid/thunx/pdp/opa/QuerySetToThunkExpressionConverter.java

```diff
@@ -279,7 +295,86 @@ public ThunkExpression<?> visit(Null nullValue) {
 
         @Override
         public ThunkExpression<?> visit(ArrayTerm arrayTerm) {
-            return null;
+            ScalarCollectingRegoVisitor collectingRegoVisitor = new ScalarCollectingRegoVisitor();
+            return Scalar.of(
+                    (List) arrayTerm.getValue().stream().map(scalarTerm -> scalarTerm.accept(collectingRegoVisitor)).collect(Collectors.toList())
```

Comment:
An `ArrayTerm`/`SetTerm` can now contain things other than scalars from OPA.

For me, it's fine to keep the restriction for arrays/sets to only contain scalars right now.

This means that an expression like `input.value1 in [input.value2, input.value3]` is not possible to express right now.

Alternatively, we accept all `ThunkExpression` in the `CollectionValue`. In that case, we don't need two separate visitors, but we can replace this with 

```suggestion
                    (List) arrayTerm.getValue().stream().map(scalarTerm -> scalarTerm.accept(this)).collect(Collectors.toList())
```

In that case, we push the decision for what things to accept further downstream.